src/sentry/integrations/utils/codecov.py

```diff
@@ -134,8 +135,15 @@ def get_codecov_data(repo: str, service: str, path: str) -> Tuple[LineCoverage |
     return line_coverage, codecov_url
 
 
-def fetch_codecov_data(config: Dict[str, Any]) -> Dict[str, Any]:
-    data = {}
+class CodecovData(TypedDict):
+    lineCoverage: NotRequired[LineCoverage]
+    coverageUrl: NotRequired[str]
+    status: NotRequired[int]
+    attemptedUrl: NotRequired[str]
+
+
+def fetch_codecov_data(config: Dict[str, Any]) -> CodecovData:
```

Comment:
Might be nice to accept `repo` `provider` and `path` instead of `config` but typing the response is already an improvement

Reply:
will try to follow up there